Brokerages/Bitfinex/BaseMessage.cs

```diff
@@ -0,0 +1,137 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Brokerages.Bitfinex
+{
+
+
+    /// <summary>
+    /// Base class for Bitfinex wss messages
+    /// </summary>
+    public abstract class BaseMessage
+    {
+
+        /// <summary>
+        /// Stored Keys for message
+        /// </summary>
+        protected string[] allKeys { get; set; }
+
+        /// <summary>
+        /// Stores values of message
+        /// </summary>
+        protected string[] allValues { get; set; }
+
+        /// <summary>
+        /// Creates base message instance
+        /// </summary>
+        /// <param name="values"></param>
+        public BaseMessage(string[] values)
+        {
+            allValues = values;
+        }
+
+        /// <summary>
+        /// Returns typed value from untyped json array
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public string GetString(string key)
+        {
+            return allValues[Array.IndexOf(allKeys, key)];
+        }
+
+        /// <summary>
+        /// Returns typed value from untyped json array
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public decimal GetDecimal(string key)
+        {
+            return decimal.Parse(allValues[Array.IndexOf(allKeys, key)]);
```

Comment:
The problem here is the data is a raw array rather than key/value pairs, so this approach avoids accessing the json array directly by index number. So, this is an attempt to get dictionary-esque lookups. The alternative would be something like:

`decimal.Parse(allValues[3])`

..which makes me shudder.